projects/memapCore/src/main/java/memap/components/ClientStorage.java

```diff
@@ -51,21 +53,48 @@
 	 * @param costCO2Id        CO2 cost [kg CO2/kWh]
 	 * @param port
 	 */
-	public ClientStorage(BasicClient client, String name, NodeId capacityId, NodeId stateOfCharge, NodeId calculatedSocId, NodeId maxChargingId,
-			NodeId maxDischargingId, NodeId effInId, NodeId effOutId, NodeId storageLossId, NodeId nodeIdSector, NodeId opCostId,
-			NodeId costCO2Id, NodeId inputSetpointsId, NodeId outputSetpointsId, int port) throws InterruptedException, ExecutionException {
-		super(name, client.readFinalDoubleValue(capacityId), client.readFinalDoubleValue(stateOfCharge),
-				client.readFinalDoubleValue(maxChargingId), client.readFinalDoubleValue(maxDischargingId),
-				client.readFinalDoubleValue(effInId), client.readFinalDoubleValue(effOutId), client.readFinalDoubleValue(storageLossId), port);
+	public ClientStorage(
+			BasicClient client, 
+			String name, 
+			NodeId capacityId, 
+			NodeId stateOfCharge, 
+			NodeId calculatedSocId, 
+			NodeId maxChargingId,
+			NodeId maxDischargingId, 
+			NodeId effInId, 
+			NodeId effOutId, 
+			NodeId storageLossId, 
+			NodeId nodeIdSector, 
+			NodeId opCostId,
+			NodeId costCO2Id, 
+			NodeId inputSetpointsId, 
+			NodeId outputSetpointsId, 
+			NodeId inputSetpointId, 
+			NodeId outputSetpointId, 
+			int port) throws InterruptedException, ExecutionException {
+		super(name, 
+				client.readFinalDoubleValue(capacityId), 
+				client.readFinalDoubleValue(stateOfCharge),
+				client.readFinalDoubleValue(maxChargingId), 
+				client.readFinalDoubleValue(maxDischargingId),
+				client.readFinalDoubleValue(effInId), 
+				client.readFinalDoubleValue(effOutId), 
+				client.readFinalDoubleValue(storageLossId), 
+				port);
 		this.client = client;
+		// setpoints as scalar and array
+		this.inputSetpointId = inputSetpointId;
 		this.inputSetpointsId = inputSetpointsId;
+		this.outputSetpointId = outputSetpointId;
 		this.outputSetpointsId = outputSetpointsId;
 		this.networkType = setNetworkType(client, nodeIdSector);
-		this.opCost = client.readFinalDoubleValue(opCostId);
-		this.costCO2 = client.readFinalDoubleValue(costCO2Id);
+//		this.opCost = client.readFinalDoubleValue(opCostId);
+//		this.costCO2 = client.readFinalDoubleValue(costCO2Id);
+		this.opCost = 0.0;
+		this.costCO2 = 0.0;

```

Comment:
In CoSES there were no datapoints for operational costs and CO2 for storage. Although CO2 was added recently, operational costs had to be set to zero here for the connection to the current CoSES Interface server to work.

Has to be changed in future versions (see #169 )